src/V3Unknown.cpp

```diff
@@ -141,14 +141,16 @@ class UnknownVisitor final : public VNVisitor {
         VL_RESTORER(m_modp);
         VL_RESTORER(m_constXCvt);
         VL_RESTORER(m_allowXUnique);
+        auto nameCountMap = std::make_unique<NameCountMap>();
```

Comment:
```suggestion
        VL_RESTORER(m_xrandNames);
```

Then shouldn't need all of the other stuff.

Reply:
Yeah, I tried this route first.  Initially I got:
```
../V3Global.h:65:26: error: use of deleted function ‘V3UniqueNames& V3UniqueNames::operator=(const V3UniqueNames&)’
   65 |     ~VRestorer() { m_ref = m_saved; }
      |                    ~~~~~~^~~~~~~~~
```
and then if I add the assignment operator it gets mad about const members.  I can get this to work, but I went the other way because using `VL_RESTORER`  will cause multiple copies of the `m_multiplicity` map which seems bad.

Do you think it's worth avoiding these copies?

Reply:
Oh, that's a good point.  Rather than making a new map, just have a pointer to V3UniqNames and make a new unique_ptr to a whole V3UniqNames, then the internals to V3UniqueNames shouldn't need to change.